test/ruby/test_time.rb

```diff
@@ -67,8 +67,12 @@ def test_new_from_string
     assert_equal(t, Time.new("2020-12-25 00:56:17 +0900"))
     assert_equal(t, Time.new("2020-12-25 00:57:47 +090130"))
     assert_equal(t, Time.new("2020-12-25T00:56:17+09:00"))
-    assert_equal(Time.utc(2020, 12, 24, 15, 56, 0), Time.new("2020-12-25 00:56 +09:00"))
```

Comment:
ISO 8601 permits hh:mm (hours and minutes but no seconds) as the extended format.

The complete extended format for local time is hh:mm:ss.
hh:mm is also defined as representations with reduced accuracy.
But hh is not defined for the extended format.

Note that, the basic format defines hhmmss, hhmm, and hh.
I don't know the reason for the difference, though.

Since the basic format and the extended format cannot be mixed,
YYYY-MM-DDThh violates ISO 8601.

I consider this method parses some part of the extended format of date and time of day of ISO 8601.

I'm not sure that Time.new should support hh:mm or not.

Reply:
>  But hh is not defined for the extended format.
> 
> Note that, the basic format defines hhmmss, hhmm, and hh. I don't know the reason for the difference, though.

That is hh is interpreted as the basic format, and the extended format conflicts with it?

> I'm not sure that Time.new should support hh:mm or not.

It isn't ambiguous and bad, I guess.